### ISSUE TITLE
Size of 100% breaks outer minimum

### ISSUE THREAD

**Issue by Enyium** [slint-ui/slint#6285]
[SlintPad demo](https://slintpad.com/?snippet=export+component+Demo+%7B%0A++++min-width%3A+200px%3B%0A++++min-height%3A+200px%3B%0A++++Rectangle+%7B%0A++++++++%2F%2F+With+just+one+line+commented+in%2C%0A++++++++%2F%2F+both+minimum+values+don%27t+work+anymore%2C%0A++++++++%2F%2F+even+though+%60100%25%60+should+be+the+default+anyways.%0A++++++++%2F%2Fwidth%3A+100%25%3B%0A++++++++%2F%2Fheight%3A+100%25%3B%0A++++%7D%0A%7D%0A&style=native); see comment:

```slint
export component Demo {
    min-width: 200px;
    min-height: 200px;
    Rectangle {
        // With just one line commented in,
        // both minimum values don't work anymore,
        // even though `100%` should be the default anyways.
        //width: 100%;
        //height: 100%;
    }
}
```

**Comment by ogoffart** [comment]
Thanks for the issue, this is indeed a bug.

**Comment by ogoffart** [comment]
I've tried to make a fix in d64e42885047c9f7893a261100fecedf77527dc5, but that actually doesn't work well as it cause a binding loop in the geometry of the DateTimePicker (for fluent and cosmic) and there is a chance that it breaks other things.
I'm not sure if the DateTimePicker is actually a loop or not, it may well be.
